pvlib/modelchain.py

```diff
@@ -260,6 +262,26 @@ def get_orientation(strategy, **kwargs):
     return surface_tilt, surface_azimuth
 
 
+@dataclass
+class ModelChainResult:
+    # system-level information
+    # weather: pd.DataFrame = field(default=None)
+    solar_position: pd.DataFrame = field(default=None)
+    airmass: pd.DataFrame = field(default=None)
+    ac: pd.Series = field(default=None)
+    # per DC array information
+    total_irrad: pd.DataFrame = field(default=None)
+    aoi: pd.Series = field(default=None)
+    aoi_modifier: pd.Series = field(default=None)
+    spectral_modifier: pd.Series = field(default=None)
+    cell_temperature: pd.Series = field(default=None)
+    effective_irradiance: pd.Series = field(default=None)
+    dc: pd.Series = field(default=None)
```

Comment:
I'm confused about the intent here... if these parameters are per DC array then shouldn't they be tuples of Series/DataFrames?

Reply:
As written those types are not correct, they were specified for the single-array version of PVSystem. We should either make them always be tuples or change the type annotations to `typing.Union[pd.Series, typing.Tuple[pd.Series, ...]]`

Reply:
That definition before the changes to pvsystem.py had converged, please correct them.

Reply:
Actually, I believe this should be the type in order to account for the default value: 

```python
class ModelChainResult:
    T = TypeVar('T')
    PerArray = Union[T, Tuple[T, ...]]
    # ...
    aio: Optional[PerArray[pd.Series]] = field(default=None)
```

I will update it.